docs/source/features/file-uploads.md

```diff
@@ -0,0 +1,134 @@
+---
+title: File Uploads
+description: Implementing File Uploads on Apollo Server
+---
+
+File uploads are frequently requested features of several applications. Apollo Server enables file uploads by default. 
+
+## Default File Uploads example
+
+To enable file uploads, reference the `Upload` type in the schema passed to the Apollo Server construction.
+
+```js
```

Comment:
Alternate example:

```js
const typeDefs = foo`
  type File {
    id: ID!
    filename: String!
    mimetype: String!
    encoding: String!
  }

  type Query {
    files: [File]
  }

  type Mutation {
    singleUpload(upload: Upload!): File!
  }
`

const resolvers = {
  Query: {
    files: () => {
      // Return record of files uploaded.
    }
  },
  Mutation: {
    async singleUpload(parent, { upload }) {
      const { stream, filename, mimetype, encoding } = await upload

      // 1. Validate file metadata.

      // 2. Stream file contents into cloud storage:
      // https://nodejs.org/api/stream.html

      // 3. Record the file upload in your DB.
      // const id = await recordFile( … )

      return { id, filename, mimetype, encoding }
    }
  }
}
```

- Added an `id` to the `File` type.
- Use async/await.
- Outline steps in resolving the upload.
- Rename the mutation argument from `file` to `uploads` for clarity as people can get confused between an `Upload` scalar and a custom `File` type.
- Also renamed the `uploads` query to `files`, as it is a `File` list.